### ISSUE TITLE
Products stock not updated

### ISSUE THREAD

**Issue by xxorax** [OCA/connector-magento#118]
I configured well the Magento backend on a fresh Odoo 8 install, and my products are sync, however the product_qty field in magento_product_product are still at zero. When I run "Update all the products stock quantities" it does nothing, the quantities still remain at 0.

I try to debug but I'm not a python dev, however I trace that new_qty in _recompute_magento_qty_backend method is always 0 for all the products.

In magento_model.update_product_stock_qty , the magento_products resultset contain well a good number of products.

Also I don't see any Magento API call when I press update qty button (and the API call work when I update categories button for example).

Do I forgot something ?

**Comment by guewen** [comment]
On Tue, Jul 14, 2015 at 8:31 PM, Martin PANEL notifications@github.com
wrote:

> I configured well the Magento backend on a fresh Odoo 8 install, and my
> products are sync, however the product_qty field in magento_product_product
> are still at zero. When I run "Update all the products stock quantities" it
> does nothing, the quantities still remain at 0.
> 
> I try to debug but I'm not a python dev, however I trace that new_qty in
> _recompute_magento_qty_backend method is always 0 for all the products.
> 
> In magento_model.update_product_stock_qty , the magento_products resultset
> contain well a good number of products.
> 
> Also I don't see any Magento API call when I press update qty button (and
> the API call work when I update categories button for example).
> 
> Do I forgot something ?
> 
> Hi,

If the quantity remains at 0, if seems like it doesn't find any quantity
for your products in Odoo.
Do you actually have stock quantities different than 0 on your products on
Odoo?
What is the warehouse configured on the Magento Backend and is it in the
same warehouse that you made inventory/receptions of your products?
What is the "stock field" configured on the Magento Backend?

**Comment by xxorax** [comment]
> If the quantity remains at 0, if seems like it doesn't find any quantity for your products in Odoo.
> Do you actually have stock quantities different than 0 on your products on Odoo?

Yes all the quantity fields in the Odoo products are at 0, even the products have been well imported from Magento, but maybe the Magento quantity in not imported well  ?

> What is the warehouse configured on the Magento Backend and is it in the
> same warehouse that you made inventory/receptions of your products?

I have only the default warehouse and I did not touch anything here. However I didn't understand the step 7 in [the backend part of the installation guide](http://odoo-magento-connector.com/guides/installation_guide.html#after-the-installation) "Create an inventory for your products"

> What is the "stock field" configured on the Magento Backend?

The stock field was "Forecast Quantity" when I imported the products. Now it is "Quantity Available" because I cannot see the field "Forecast Quantity" when I edit an Odoo product.

I just created a new product in Magento with a quantity, and I run the import product in Odoo again to see if the problem come from the quantity field, but it still remains the same with the quantity field "Quantity Available" : all the quantity fields in Odoo product is still 0.

Thanks for your help

**Comment by xxorax** [comment]
Also, I have an other question: How Odoo get the stock changes from Magento ? Update quantity button in the backend does not make any Magento API call so is it done only by a Scheduled Action ?
Thanks

**Comment by guewen** [comment]
Not sure I understood, do you expect to see the Magento quantity imported
into Odoo?
It is the other way around: the master of the stock is Odoo. Once you have
some quantities in Odoo you will be able to export them to Magento.

**Comment by xxorax** [comment]
Ok, I see. Maybe it needs to be clearly write in the doc, I did not see that.

And so, how Odoo get the stock changes from Magento ? When there is a Magento order and when we don't import the orders/invoices ?
